src/neuroconv/tools/testing/mock_ttl_signals.py

```diff
@@ -0,0 +1,290 @@
+from numpy.typing import DTypeLike
+from nwbinspector.tools import make_minimal_nwbfile
+from nwbinspector.utils import is_module_installed
+from pynwb import NWBHDF5IO, TimeSeries, H5DataIO
+
+from ...utils import IntType, FloatType, ArrayType, FolderPathType
+
+
+def generate_mock_ttl_signal(
+    signal_duration: float = 5.5,
+    sampling_frequency_hz: float = 25000.0,
+    ttl_on_duration: float = 1.0,
+    ttl_off_duration: Optional[float] = None,
+    ttl_times: Optional[ArrayType] = None,
+    dtype: DTypeLike = "int16",
+    baseline_mean: Optional[Union[IntType, FloatType]] = None,
+    signal_mean: Optional[Union[IntType, FloatType]] = None,
+    channel_noise: Optional[Union[IntType, FloatType]] = None,
+    random_seed: Optional[int] = 0,
+) -> np.ndarray:
+    """
+    Generate a synthetic signal of TTL pulses similar to those seen in .nidq.bin files using SpikeGLX.
+
+    Parameters
+    ----------
+    signal_duration: float, optional
+        The number of seconds to simulate.
+        The default is 5.5 seconds.
+    sampling_frequency_hz: float, optional
+        The sampling frequency of the signal in Hz.
+        The default is 25000 Hz; similar to that of .nidq.bin files.
+    ttl_on_duration: float, optional
+        How long the TTL pulse stays in the 'on' state when triggered.
+        The default is 1 second to match the default frequency of 2 seconds to emulate a regular metronomic pulse.
+    ttl_off_duration: float, optional
+        How long the TTL pulse stays in the 'off' state when triggered.
+        The default is 1 second to match the default frequency of 2 seconds to emulate a regular metronomic pulse.
+    ttl_times: array of floats, optional
+        If an irregular series of TTL pulses are desired, specify the exact sequence of timings.
+        The default assumes regular series with the `ttl_frequency_hz` rate.
+    dtype: numpy data type or one of its accepted string input
+        The data type of the trace and its specifiable parameters.
+        Recommended to be int16 for maximum efficiency, but can also be any size float to represent voltage scalings.
+        Defaults to int16.
+    baseline_mean: integer or float, depending on specified 'dtype', optional
+        The average value for the baseline; usually around 0 Volts.
+        The default is apprimxately 0.005645752 Volts, estimated from a real example of a TTL pulse in a .nidq.bin file.
+    signal_mean: integer or float, depending on specified 'dtype', optional
+        The average value for the signal; usually around 5 Volts.
+        The default is apprimxately 4.980773925 Volts, estimated from a real example of a TTL pulse in a .nidq.bin file.
+    channel_noise: integer or float, depending on specified 'dtype', optional
+        The average value for the noise of the channel.
+        The default is apprimxately 0.002288818 Volts, estimated from a real example of a TTL pulse in a .nidq.bin file.
+    random_seed: int or None, optional
```

Comment:
```suggestion
    random_seed: int, optional
```

optional means that None is an option

Reply:
That's what `typing.Optional` annotation means, but Numpy docstring format just means that the argument is not required: https://numpydoc.readthedocs.io/en/latest/format.html#parameters

Reply:
In particular, I include the `None` explicitly here to inform that's how you disable seed-fixing the RNG, which is the stated default for reproducibility.